operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -798,10 +819,14 @@ func (r *StackReconciler) markStackFailed(sess *StackReconcilerSession, instance
 		LastAttemptedCommit: current.Commit,
 		Permalink:           shared.Permalink(update.Status.Permalink),
 		LastResyncTime:      metav1.Now(),
+		Attempts:            1,
 	}
 	if last != nil {
 		instance.Status.LastUpdate.LastSuccessfulCommit = last.LastSuccessfulCommit
 	}
+	if last != nil && last.LastAttemptedCommit == current.Commit {
```

Comment:
May I suggest the following logic, to mean that we reset the atttempts whenever there's a generation change (as seen in big the `synced` expression).  Your expression wouldn't reset the cooldown in the face of a configuration property change.

```go
if last != nil && last.Generation == current.Generation {
   instance.Status.LastUpdate.Attempts += last.Attempts
}

Reply:
Yep that's a good point!